exec/java-exec/src/main/java/org/apache/drill/exec/server/options/TypeValidators.java

```diff
@@ -179,6 +181,7 @@ public void validate(final OptionValue v) {
 
     public TypeValidator(final String name, final Kind kind, final OptionValue defValue) {
       super(name);
+      checkArgument(defValue.type == OptionType.SYSTEM, "Default value must be SYSTEM type.");
```

Comment:
Can you modify all of these to return UserException? Maybe create a static method ideally reporting the specific option and value as additional UserException context.